homeassistant/components/tado/__init__.py

```diff
@@ -41,45 +44,51 @@
 
 def setup(hass, config):
     """Set up of the Tado component."""
-    username = config[DOMAIN][CONF_USERNAME]
-    password = config[DOMAIN][CONF_PASSWORD]
+    acc_list = config[DOMAIN]
 
-    tadoconnector = TadoConnector(hass, username, password)
-    if not tadoconnector.setup():
-        return False
+    api_data_list = []
 
-    hass.data[DOMAIN] = tadoconnector
+    for acc in acc_list:
+        username = acc[CONF_USERNAME]
+        password = acc[CONF_PASSWORD]
+        fallback = acc[CONF_FALLBACK]
 
-    # Do first update
-    tadoconnector.update()
+        tadoconnector = TadoConnector(hass, username, password, fallback)
+        if not tadoconnector.setup():
+            continue
+
+        # Do first update
+        tadoconnector.update()
+
+        api_data_list.append(tadoconnector)
+        # Poll for updates in the background
+        hass.helpers.event.track_time_interval(
+            lambda now, tc=tadoconnector: tc.update(), SCAN_INTERVAL
```

Comment:
This won't work. `tadoconnector` will always point at the latest defined one, as Lambdas do not maintain their own variable scope but use the one where they are defined.

Reply:
Also what are you doing with default vars anyway here.

Reply:
AFAIU if I use ```tadoconnector``` direct, I should get a closure with latest defined object. When I use it as parameter, I do not get a closure but pass current object into lambda.

Reply:
oh it's a hack to work around the issue with closures. Please add a comment.

Reply:
Done. I'm only not sure that my comment ist understandable.